test/test_gpu_view.cpp

```diff
@@ -2478,4 +2478,53 @@ TEST_F(GpuViewTest, ExpandedBroadcast) {
       __FILE__);
 }
 
+// See https://github.com/NVIDIA/Fuser/issues/1126
+TEST_F(GpuViewTest, SplitExpandedBroadcast) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* in = makeContigConcreteTensor({4, 5});
+  fusion.addInput(in);
+  TensorView* out = broadcast(in, {false, false, true});
+  out = expand(
+      out,
+      {IrBuilder::create<Val>(4),
+       IrBuilder::create<Val>(5),
+       IrBuilder::create<Val>(6)});
+  out = reshape(out, {4, 5, 6}, {4, 5, 2, 3});
+  fusion.addOutput(out);
+
+  FusionExecutor fe;
+  at::Tensor in_tensor =
+      at::randn({4, 5}, at::dtype(at::kFloat).device(at::kCUDA, 0));
+  fe.compileFusion(&fusion, {in_tensor});
+  at::Tensor out_tensor = fe.runFusion({in_tensor})[0];
+  EXPECT_THAT(out_tensor.strides(), testing::ElementsAre(5, 1, 0, 0));
+}
+
+// Reverse of SplitExpandedBroadcast
+TEST_F(GpuViewTest, MergeExpandedBroadcast) {
```

Comment:
Also test merging a broadcast and a non-broadcast?

Reply:
Done.